src/fetchData/util.ts

```diff
@@ -0,0 +1,43 @@
+import { resourceNameArray, Resources } from '../utils/response/apiResponse';
+import { RawApiResponse } from '../utils/response/rawApiResponse';
+import { ResourceResponse } from '../utils/response/resourceResponse';
+import { FetchDataResponse } from './type';
+
+export function createFetchDataResponse(rawApiResponse: RawApiResponse): FetchDataResponse {
+  const results = {} as Resources;
+  resourceNameArray.forEach((resource) => {
+    const clarifiedKey = resource as keyof Resources;
+    const resourceResponse = convert(rawApiResponse, resource);
+    if (resourceResponse) {
+      Object.defineProperty(results, clarifiedKey, resourceResponse);
+    }
+  });
+  return {
+    success: true,
+    apiResponse: rawApiResponse,
+    results,
+  };
+}
+
+function convert<T>(fetchDataResponse: RawApiResponse, resource: keyof Resources): ResourceResponse<T> | undefined {
```

Comment:
Simply avoided to use the one in `deltaSync` as it uses a different type and aimed to avoid code complexity. Should we find the need to change anything in any of these areas in future, we will have much calmer endings this way.